### DISCUSSION TITLE
Trigger requests based on conditions

### DISCUSSION THREAD

**Discussion by ClausClaus** [TanStack/query#1075] [General]
👋Hello! I found that this library defines the request configuration at the top of the component in a declarative way. I would like to ask, if I have a pop-up window, I want to initiate the request and load the data after opening the pop-up window. This can be configured in useEffect. Monitor the visible variable of the pop-up window to achieve the goal, but what to do in useQuery! I need help, hope someone can tell me, I apologize for the inconvenience caused to you 🙏...

**Comment by denisborovikov** [comment]
This might help https://react-query.tanstack.com/docs/guides/queries#dependent-queries. Use the `enabled` option to initiate your query as it ready. Also check https://react-query.tanstack.com/docs/api#usequery
